sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java

```diff
@@ -0,0 +1,151 @@
+    private final long cacheRefreshIntervalInMs;
+    private final List<String> secretKeys;
+
+    private final Object refreshLock = new Object();
+    private final SecretClient keyVaultClient;
+    private final String vaultUri;
+
+    private ArrayList<String> propertyNames = new ArrayList<>();
+    private String[] propertyNamesArr;
+
+    private final AtomicLong lastUpdateTime = new AtomicLong();
+    private final ReadWriteLock rwLock = new ReentrantReadWriteLock();
+
+    public KeyVaultOperation(final SecretClient keyVaultClient,
+                             String vaultUri,
+                             final long refreshInterval,
+                             final List<String> secretKeys) {
+        this.cacheRefreshIntervalInMs = refreshInterval;
+        this.secretKeys = secretKeys;
+        this.keyVaultClient = keyVaultClient;
+        // TODO(pan): need to validate why last '/' need to be truncated.
+        this.vaultUri = StringUtils.trimTrailingCharacter(vaultUri.trim(), '/');
+        fillSecretsList();
+    }
+
+    public String[] list() {
+        try {
+            this.rwLock.readLock().lock();
+            return propertyNamesArr;
+        } finally {
+            this.rwLock.readLock().unlock();
+        }
+    }
+
+    private String getKeyVaultSecretName(@NonNull String property) {
+        if (property.matches("[a-z0-9A-Z-]+")) {
+            return property.toLowerCase(Locale.US);
+        } else if (property.matches("[A-Z0-9_]+")) {
+            return property.toLowerCase(Locale.US).replaceAll("_", "-");
+        } else {
+            return property.toLowerCase(Locale.US)
+                    .replaceAll("-", "")     // my-project -> myproject
+                    .replaceAll("_", "")     // my_project -> myproject
+                    .replaceAll("\\.", "-"); // acme.myproject -> acme-myproject
+        }
+    }
+
+    /**
+     * For convention we need to support all relaxed binding format from spring, these may include:
+     * <ul>
+     * <li>Spring relaxed binding names</li>
+     * <li>acme.my-project.person.first-name</li>
+     * <li>acme.myProject.person.firstName</li>
+     * <li>acme.my_project.person.first_name</li>
+     * <li>ACME_MYPROJECT_PERSON_FIRSTNAME</li>
+     * </ul>
+     * But azure keyvault only allows ^[0-9a-zA-Z-]+$ and case insensitive, so there must be some conversion
+     * between spring names and azure keyvault names.
+     * For example, the 4 properties stated above should be convert to acme-myproject-person-firstname in keyvault.
+     *
+     * @param property of secret instance.
+     * @return the value of secret with given name or null.
+     */
+    public String get(final String property) {
+        Assert.hasText(property, "property should contain text.");
+        final String secretName = getKeyVaultSecretName(property);
+
+        //if user don't set specific secret keys, then refresh token
+        if (this.secretKeys == null || secretKeys.size() == 0) {
```

Comment:
Use `this` consistently. Here one is using `this` and another is not: `this.secretKeys` and `secretKeys`

Reply:
done.